File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        li = self.inOrder(root, [])
        return li[k-1]

    def inOrder(self, r, l):
        if(r==None):
            return l
        
        self.inOrder(r.left, l)
        l.append(r.val)
        self.inOrder(r.right, l)
        return l

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int kthSmallest(TreeNode root, int k) {
        List<Integer> li = inOrderTraversal(root, new ArrayList<>());
        return li.get(k-1);
    }
    public static List<Integer> inOrderTraversal(TreeNode r, List<Integer> l){
        if(r==null){return l;}
        inOrderTraversal(r.left, l);
        l.add(r.val);
        inOrderTraversal(r.right, l);
        return l;
    }
}